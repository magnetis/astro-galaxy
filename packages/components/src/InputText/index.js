import { IconAlert, IconCircleCheck } from '@magnetis/astro-galaxy-icons';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { layout, space } from 'styled-system';

export const inputSizes = {
  medium: 'medium',
  large: 'large',
};

const inputSizeStyles = props => {
  if (props.inputSize === inputSizes.large) {
    return {
      label: {
        initial: `
          font-size: ${props.theme.fontSizes.texts.large};
          padding: 0.6em;
        `,
        focus: `transform: translate(8px, -3px) scale(0.6);`,
      },
      input: `font-size: ${props.theme.fontSizes.texts.large};`,
    };
  }

  return {
    label: {
      initial: `font-size: ${props.theme.fontSizes.texts.medium};`,
      focus: `transform: translate(3px, -6px) scale(0.8);`,
    },
  };
};

const labelValidationStyles = props => {
  if (props.isValidated && !props.disabled) {
    return `color: ${props.theme.colors.earth400}`;
  }

  if (props.isInvalid && !props.disabled) {
    return `color: ${props.theme.colors.mars500}`;
  }

  if (props.disabled) {
    return `
      color: ${props.theme.colors.moon300};
      cursor: not-allowed;
    `;
  }

  return `color: ${props.defaultColor}`;
};

const inputValidationStyles = props => {
  const iconPadding = props.theme.space[6];

  if (props.isValidated && !props.disabled) {
    return `border-color: ${props.theme.colors.earth400}; padding-right: ${iconPadding}px;`;
  }

  if (props.isInvalid && !props.disabled) {
    return `border-color: ${props.theme.colors.mars500}; padding-right: ${iconPadding}px;`;
  }

  if (props.disabled) {
    return `
      color: ${props.theme.colors.moon300};
      cursor: not-allowed;
      background-color: ${props.theme.colors.moon100};
      border-color:  ${props.theme.colors.moon200};
    `;
  }

  return `border-color: ${props.defaultColor}`;
};

const InputWrapper = styled.fieldset`
  border: 0;
  padding: 0;
  position: relative;

  ${space};
  ${layout};
`;

const InputLabel = styled.label`
  font: ${props => props.theme.fonts.secondary};
  height: 4em;
  padding: 1em;
  pointer-events: none;
  user-select: none;

  ${props => labelValidationStyles({ ...props, defaultColor: props.theme.colors.moon400 })};
  ${props => inputSizeStyles(props).label.initial};
`;

const InputField = styled.input`
  background-color: ${props => props.theme.colors.space100};
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: ${props => props.theme.colors.moon900};
  font: ${props => props.theme.fonts.secondary};
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;

  ${props => inputValidationStyles({ ...props, defaultColor: props.theme.colors.moon500 })};
  ${props => inputSizeStyles(props).input};

  &::placeholder {
    color: transparent;
  }

  &:focus::placeholder,
  &.--no-animate::placeholder {
    color: ${props => props.theme.colors.moon300};
  }

  &:focus {
    outline: none;

    ${props => inputValidationStyles({ ...props, defaultColor: props.theme.colors.moon500 })};
  }

  &:focus + label {
    ${props => labelValidationStyles({ ...props, defaultColor: props.theme.colors.uranus500 })};
  }

  &:focus,
  &:not(:placeholder-shown),
  &.--no-animate {
    & + label {
      font-weight: 700;

      ${props => inputSizeStyles(props).label.focus};
    }
    ${props => inputValidationStyles({ ...props, defaultColor: props.theme.colors.uranus500 })};
  }

  & + label {
    left: 0;
    position: absolute;
    top: 0;
    transition: transform 0.25s, opacity 0.25s ease-in-out;
    transform-origin: 0 0;
  }
`;

const iconStyles = inputSize => ({
  position: 'absolute',
  top: `${inputSize === inputSizes.large ? 16 : 10}px`,
  right: '8px',
});

const ErrorMessage = styled.div`
  color: ${props => props.theme.colors.mars500};
  font: ${props => props.theme.fonts.secondary};
  font-size: ${props => props.theme.fontSizes.texts.verySmall};
  margin: 4px 16px;
`;

const InputText = ({
  inputId,
  labelId,
  labelText,
  isValidated,
  isInvalid,
  errorMessage,
  labelProps,
  noAnimate,
  className,
  inputSize,
  ...rest
}) => (
  <>
    <InputWrapper>
      <InputField
        type="text"
        id={inputId}
        aria-labelledby={labelId}
        placeholder={labelText}
        isValidated={isValidated}
        isInvalid={isInvalid}
        inputSize={inputSize}
        className={`${className} ${noAnimate ? '--no-animate' : ''}`}
        {...rest}
      />
      <InputLabel
        id={labelId}
        htmlFor={inputId}
        isValidated={isValidated}
        isInvalid={isInvalid}
        inputSize={inputSize}
        {...labelProps}>
        {labelText}
      </InputLabel>
      {isValidated && (
        <IconCircleCheck color="earth400" size="32" style={{ ...iconStyles(inputSize) }} />
      )}
      {isInvalid && <IconAlert color="mars500" size="32" style={{ ...iconStyles(inputSize) }} />}
      {isInvalid && errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </InputWrapper>
  </>
);

InputText.displayName = 'InputText';
ErrorMessage.displayName = 'ErrorMessage';

InputText.defaultProps = {
  labelText: 'Standard',
  isValidated: false,
  isInvalid: false,
  noAnimate: false,
  labelProps: {},
  className: '',
  inputSize: inputSizes.medium,
};

InputText.propTypes = {
  inputId: PropTypes.string.isRequired,
  labelId: PropTypes.string.isRequired,
  labelText: PropTypes.string,
  isValidated: PropTypes.bool,
  isInvalid: PropTypes.bool,
  noAnimate: PropTypes.bool,
  errorMessage: PropTypes.string,
  labelProps: PropTypes.object,
  inputSize: PropTypes.oneOf([inputSizes.medium, inputSizes.large]),
};

export default InputText;
