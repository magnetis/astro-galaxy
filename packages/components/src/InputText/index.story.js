import { boolean, radios, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import InputText, { inputSizes } from './index.js';

const buildProps = (props = {}) => ({
  inputId: 'inputId',
  labelId: 'labelId',
  labelText: text('label', 'Label'),
  placeholder: text('placeholder', 'Placeholder'),
  inputSize: radios('size', inputSizes, inputSizes.medium),
  noAnimate: boolean('no animate', false),
  isValidated: boolean('is validated', false),
  isInvalid: boolean('is invalid', false),
  errorMessage: text('error message', 'Invalid data'),
  disabled: boolean('disabled', false),
  ...props,
});

storiesOf('inputs|standard text inputs', module)
  .addDecorator(withKnobs)
  .add('default', () => <InputText {...buildProps()} />)
  .add('validated', () => (
    <InputText
      {...buildProps({
        labelText: text('label', 'Validated'),
        isValidated: boolean('is validated', true),
      })}
      defaultValue="Astro Team"
    />
  ))
  .add('invalid', () => (
    <InputText
      {...buildProps({
        labelText: text('label', 'Invalid'),
        isInvalid: boolean('is invalid', true),
      })}
      defaultValue="XYZK2"
    />
  ))
  .add('disabled', () => (
    <InputText
      {...buildProps({
        disabled: boolean('disabled', true),
      })}
    />
  ))
  .add('without animation', () => (
    <InputText
      {...buildProps({
        noAnimate: boolean('no animate', true),
      })}
    />
  ))
  .add('large size', () => (
    <InputText
      {...buildProps({
        inputSize: radios('size', inputSizes, inputSizes.large),
      })}
    />
  ));
