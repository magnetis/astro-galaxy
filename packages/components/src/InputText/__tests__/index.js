import { webTheme } from '@magnetis/astro-galaxy-themes';
import React from 'react';
import InputText, { inputSizes } from '../index';

function mount(props = {}) {
  const mergedProps = {
    inputId: 'inputId',
    labelId: 'labelId',
    ...props,
  };
  return rendererCreateWithTheme(<InputText {...mergedProps} />).toJSON();
}

describe('input text', () => {
  it('renders correctly the standard input', () => {
    const wrapper = mount();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly the validated input', () => {
    const wrapper = mount({
      isValidated: true,
      labelText: 'Validated',
      defaultValue: 'Astro Team',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly the invalid input', () => {
    const wrapper = mount({
      isInvalid: true,
      labelText: 'Invalid',
      defaultValue: 'Astro Team',
      errorMessage: 'Invalid data',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly the input without animation', () => {
    const wrapper = mount({ noAnimate: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly the large input and checks its font size', () => {
    const wrapper = mount({ inputSize: inputSizes.large });
    const input = wrapper.children[0];
    expect(input).toHaveStyleRule('font-size', webTheme.fontSizes.texts.large);
    expect(wrapper).toMatchSnapshot();
  });
});
