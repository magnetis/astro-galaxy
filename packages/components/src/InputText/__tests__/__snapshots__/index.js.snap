// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text renders correctly the input without animation 1`] = `
.c0 {
  border: 0;
  padding: 0;
  position: relative;
}

.c2 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #8296a4;
  font-size: 16px;
}

.c1 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #597183;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus::-webkit-input-placeholder,
.c1.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c1:focus::-moz-placeholder,
.c1.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c1:focus:-ms-input-placeholder,
.c1.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c1:focus::placeholder,
.c1.--no-animate::placeholder {
  color: #afbec9;
}

.c1:focus {
  outline: none;
  border-color: #597183;
}

.c1:focus + label {
  color: #159ce4;
}

.c1:focus,
.c1:not(:placeholder-shown),
.c1.--no-animate {
  border-color: #159ce4;
}

.c1:focus + label,
.c1:not(:placeholder-shown) + label,
.c1.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(3px,-6px) scale(0.8);
  -ms-transform: translate(3px,-6px) scale(0.8);
  transform: translate(3px,-6px) scale(0.8);
}

.c1.--no-animate {
  border-color: #597183;
}

.c1.--no-animate:focus {
  border-color: #159ce4;
}

.c1 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

<fieldset
  className="c0"
>
  <input
    aria-labelledby="labelId"
    className="c1  --no-animate"
    id="inputId"
    placeholder="Standard"
    type="text"
  />
  <label
    className="c2"
    htmlFor="inputId"
    id="labelId"
  >
    Standard
  </label>
</fieldset>
`;

exports[`input text renders correctly the invalid input 1`] = `
.c3 {
  fill: currentcolor;
  color: #d45459;
}

.c0 {
  border: 0;
  padding: 0;
  position: relative;
}

.c2 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #d45459;
  font-size: 16px;
}

.c1 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #d45459;
  padding-right: 48px;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus::-webkit-input-placeholder,
.c1.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c1:focus::-moz-placeholder,
.c1.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c1:focus:-ms-input-placeholder,
.c1.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c1:focus::placeholder,
.c1.--no-animate::placeholder {
  color: #afbec9;
}

.c1:focus {
  outline: none;
  border-color: #d45459;
  padding-right: 48px;
}

.c1:focus + label {
  color: #d45459;
}

.c1:focus,
.c1:not(:placeholder-shown),
.c1.--no-animate {
  border-color: #d45459;
  padding-right: 48px;
}

.c1:focus + label,
.c1:not(:placeholder-shown) + label,
.c1.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(3px,-6px) scale(0.8);
  -ms-transform: translate(3px,-6px) scale(0.8);
  transform: translate(3px,-6px) scale(0.8);
}

.c1.--no-animate {
  border-color: #d45459;
  padding-right: 48px;
}

.c1.--no-animate:focus {
  border-color: #d45459;
  padding-right: 48px;
}

.c1 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c4 {
  color: #d45459;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  font-size: 12px;
  margin: 4px 16px;
}

<fieldset
  className="c0"
>
  <input
    aria-labelledby="labelId"
    className="c1  "
    defaultValue="Astro Team"
    id="inputId"
    placeholder="Invalid"
    type="text"
  />
  <label
    className="c2"
    htmlFor="inputId"
    id="labelId"
  >
    Invalid
  </label>
  <svg
    className="c3"
    color="mars500"
    height="32"
    style={
      Object {
        "position": "absolute",
        "right": "8px",
        "top": "10px",
      }
    }
    viewBox="0 0 32 32"
    width="32"
  >
    <defs>
      <path
        d="M16.5 25.7a9.2 9.2 0 1 0 0-18.4 9.2 9.2 0 0 0 0 18.4zm0 2.3C10.149 28 5 22.851 5 16.5S10.149 5 16.5 5 28 10.149 28 16.5 22.851 28 16.5 28zm0-13.8c.635 0 1.15.515 1.15 1.15v5.75a1.15 1.15 0 1 1-2.3 0v-5.75c0-.635.515-1.15 1.15-1.15zm0-1.15a1.15 1.15 0 1 1 0-2.3 1.15 1.15 0 0 1 0 2.3z"
        id="icon-alert-a"
      />
    </defs>
    <g
      id="icon/secondary/alert"
    >
      <mask
        id="mask-2"
      >
        <use
          xlinkHref="#icon-alert-a"
        />
      </mask>
      <use
        id="Combined-Shape"
        transform="rotate(-180 16.5 16.5)"
        xlinkHref="#icon-alert-a"
      />
    </g>
  </svg>
  <div
    className="c4"
  >
    Invalid data
  </div>
</fieldset>
`;

exports[`input text renders correctly the large input and checks its font size 1`] = `
.c0 {
  border: 0;
  padding: 0;
  position: relative;
}

.c2 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #8296a4;
  font-size: 24px;
  padding: 0.6em;
}

.c1 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #597183;
  font-size: 24px;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus::-webkit-input-placeholder,
.c1.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c1:focus::-moz-placeholder,
.c1.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c1:focus:-ms-input-placeholder,
.c1.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c1:focus::placeholder,
.c1.--no-animate::placeholder {
  color: #afbec9;
}

.c1:focus {
  outline: none;
  border-color: #597183;
}

.c1:focus + label {
  color: #159ce4;
}

.c1:focus,
.c1:not(:placeholder-shown),
.c1.--no-animate {
  border-color: #159ce4;
}

.c1:focus + label,
.c1:not(:placeholder-shown) + label,
.c1.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(8px,-3px) scale(0.6);
  -ms-transform: translate(8px,-3px) scale(0.6);
  transform: translate(8px,-3px) scale(0.6);
}

.c1.--no-animate {
  border-color: #597183;
}

.c1.--no-animate:focus {
  border-color: #159ce4;
}

.c1 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

<fieldset
  className="c0"
>
  <input
    aria-labelledby="labelId"
    className="c1  "
    id="inputId"
    placeholder="Standard"
    type="text"
  />
  <label
    className="c2"
    htmlFor="inputId"
    id="labelId"
  >
    Standard
  </label>
</fieldset>
`;

exports[`input text renders correctly the standard input 1`] = `
.c0 {
  border: 0;
  padding: 0;
  position: relative;
}

.c2 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #8296a4;
  font-size: 16px;
}

.c1 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #597183;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus::-webkit-input-placeholder,
.c1.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c1:focus::-moz-placeholder,
.c1.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c1:focus:-ms-input-placeholder,
.c1.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c1:focus::placeholder,
.c1.--no-animate::placeholder {
  color: #afbec9;
}

.c1:focus {
  outline: none;
  border-color: #597183;
}

.c1:focus + label {
  color: #159ce4;
}

.c1:focus,
.c1:not(:placeholder-shown),
.c1.--no-animate {
  border-color: #159ce4;
}

.c1:focus + label,
.c1:not(:placeholder-shown) + label,
.c1.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(3px,-6px) scale(0.8);
  -ms-transform: translate(3px,-6px) scale(0.8);
  transform: translate(3px,-6px) scale(0.8);
}

.c1.--no-animate {
  border-color: #597183;
}

.c1.--no-animate:focus {
  border-color: #159ce4;
}

.c1 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

<fieldset
  className="c0"
>
  <input
    aria-labelledby="labelId"
    className="c1  "
    id="inputId"
    placeholder="Standard"
    type="text"
  />
  <label
    className="c2"
    htmlFor="inputId"
    id="labelId"
  >
    Standard
  </label>
</fieldset>
`;

exports[`input text renders correctly the validated input 1`] = `
.c3 {
  fill: currentcolor;
  color: #74e24a;
}

.c0 {
  border: 0;
  padding: 0;
  position: relative;
}

.c2 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #74e24a;
  font-size: 16px;
}

.c1 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #74e24a;
  padding-right: 48px;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus::-webkit-input-placeholder,
.c1.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c1:focus::-moz-placeholder,
.c1.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c1:focus:-ms-input-placeholder,
.c1.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c1:focus::placeholder,
.c1.--no-animate::placeholder {
  color: #afbec9;
}

.c1:focus {
  outline: none;
  border-color: #74e24a;
  padding-right: 48px;
}

.c1:focus + label {
  color: #74e24a;
}

.c1:focus,
.c1:not(:placeholder-shown),
.c1.--no-animate {
  border-color: #74e24a;
  padding-right: 48px;
}

.c1:focus + label,
.c1:not(:placeholder-shown) + label,
.c1.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(3px,-6px) scale(0.8);
  -ms-transform: translate(3px,-6px) scale(0.8);
  transform: translate(3px,-6px) scale(0.8);
}

.c1.--no-animate {
  border-color: #74e24a;
  padding-right: 48px;
}

.c1.--no-animate:focus {
  border-color: #74e24a;
  padding-right: 48px;
}

.c1 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

<fieldset
  className="c0"
>
  <input
    aria-labelledby="labelId"
    className="c1  "
    defaultValue="Astro Team"
    id="inputId"
    placeholder="Validated"
    type="text"
  />
  <label
    className="c2"
    htmlFor="inputId"
    id="labelId"
  >
    Validated
  </label>
  <svg
    className="c3"
    color="earth400"
    height="32"
    style={
      Object {
        "position": "absolute",
        "right": "8px",
        "top": "10px",
      }
    }
    viewBox="0 0 32 32"
    width="32"
  >
    <defs>
      <path
        d="M15.51 18.52l5.992-6.682a1 1 0 1 1 1.49 1.335l-6.698 7.468a1 1 0 0 1-1.452.04l-3.957-3.957a1 1 0 0 1 1.414-1.414l3.21 3.21zM16 28.5C9.096 28.5 3.5 22.904 3.5 16S9.096 3.5 16 3.5 28.5 9.096 28.5 16 22.904 28.5 16 28.5zm0-2c5.799 0 10.5-4.701 10.5-10.5S21.799 5.5 16 5.5 5.5 10.201 5.5 16 10.201 26.5 16 26.5z"
        id="icon-circle-check-a"
      />
    </defs>
    <g
      id="icon/secondary/circle/check"
    >
      <mask
        id="icon-circle-check-b"
      >
        <use
          xlinkHref="#icon-circle-check-a"
        />
      </mask>
      <use
        id="Mask"
        xlinkHref="#icon-circle-check-a"
      />
      <g
        id="color/moon-900"
        mask="url(#icon-circle-check-b)"
      >
        <path
          d="M0 0h32v32H0z"
          id="Rectangle"
        />
      </g>
    </g>
  </svg>
</fieldset>
`;
